### PR TITLE
Ensure that "null" can only be specified in "set-" commands

### PR DESCRIPTION
**Pull request by jefferai**
This allows a check function to be set to determine whether it is
allowed to be set. This also ensures that if null is allowed it is
either the only value or not set.

### CODE REVIEW

File: internal/cmd/base/flags.go [hunk 1]
```diff
@@ -581,25 +586,53 @@ func (f *FlagSet) StringSliceVar(i *StringSliceVar) {
...
+			// error; we don't check if _all_ are "null" since presumably this
+			// function prevents there being more values than just "null" if
+			// that's specified
+			if len(*s.target) > 0 && (*s.target)[0] != "null" {
+				return fmt.Errorf(`"null" cannot be combined with other values`)
+			}
+			// Set the target to only contain "null" and return
+			*s.target = []string{"null"}
+			return nil
+			// Something came in that isn't "null" but we already have "null"
```
Comment (A-440Hz) [inline internal/cmd/base/flags.go:629]: nit: can we move this comment back one tab or start it on a new line? (unless this way is already an established precedent)

**Review by A-440Hz — APPROVED** [review]
lgtm!

File: internal/cmd/base/flags.go (hunk 1 above)
Comment (jefferai) [reply internal/cmd/base/flags.go:629]: Yeah, I had done that and didn't notice that autoformatting pulled it forward. I moved it down into the if clause itself.

**Review by mgaffney — APPROVED** [review]
Nice!

**Comment by jefferai** [comment]
Merging for now; this can be rolled out in bits.

**Pull request merged**